internal/scaffold/olm-catalog/csv.go

```diff
@@ -275,6 +294,9 @@ func getEmptyRequiredCSVFields(csv *olmapiv1alpha1.ClusterServiceVersion) (field
 	if csv.Spec.Maturity == "" {
 		fields = append(fields, "spec.maturity")
 	}
+	if len(csv.Spec.Icon) == 0 || (len(csv.Spec.Icon[0].Data) == 0 && len(csv.Spec.Icon[0].MediaType) == 0) {
```

Comment:
I wouldn't add any "required" field checks here since:

1. A lot of these fields aren't required, especially `spec.icon`.
1. We should use the API validation library instead of doing this check ourselves.

2 should be done in a follow-up PR.

Reply:
It is just to show the info in the WARN.

Reply:
Setting blank default icon fields is sufficient to suggest usage. Warning a user about a non-required field is not good UX, which is why we need to switch to the API validation library.

#2082 wants `spec.icon` to be sorted to the bottom of the CSV manifest so verbose icon data doesn't clutter your screen, which I'm ok with doing but this PR does not do that.

Reply:
Thank for the help.